File: 02_ProgrammingFundamentals/_02_DataTypesAndVariables/_01_Lecture/_12_RefactorSpecialNumbers.java
package _02_ProgrammingFundamentals._02_DataTypesAndVariables._01_Lecture;

import java.util.Scanner;

public class _12_RefactorSpecialNumbers {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        int n = Integer.parseInt(scan.nextLine());

        // Find special number
        for (int i = 1; i <= n; i++) {
            int num = i;
            int sum = 0;

            while (num != 0) {
                sum += num % 10;
                num /= 10;
            }

            // Print result
            if (sum  == 5 || sum  == 7 || sum == 11) {
                System.out.printf("%d -> True%n", i);
            } else {
                System.out.printf("%d -> False%n", i);
            }
        }

        scan.close();
    }
}

File: 24_C#Fundamentals/_02_DataTypesAndVariables/_01_Lecture/_12_RefactorSpecialNumbers/_12_RefactorSpecialNumbers.cs
using System;

public class _12_RefactorSpecialNumbers
{
    public static void Main(string[] args)
    {
        // Read input value
        int n = int.Parse(Console.ReadLine());

        // Find special number
        for (int i = 1; i <= n; i++)
        {
            int num = i;
            int sum = 0;

            while (num != 0)
            {
                sum += num % 10;
                num /= 10;
            }

            // Print result
            if (sum == 5 || sum == 7 || sum == 11)
            {
                Console.WriteLine($"{i} -> True");
            }
            else
            {
                Console.WriteLine($"{i} -> False");
            }
        }
    }
}
